diffusion/models/models.py

```diff
@@ -181,39 +199,34 @@ def stable_diffusion_xl(
             metric.requires_grad_(False)
 
     if pretrained:
-        raise NotImplementedError('Full SDXL pipeline not implemented yet.')
+        unet = UNet2DConditionModel.from_pretrained(unet_model_name, subfolder='unet')
     else:
         config = PretrainedConfig.get_config_dict(unet_model_name, subfolder='unet')
-        # Currently not doing micro-conditioning, so set config appropriately
-        config[0]['addition_embed_type'] = None
-        config[0]['cross_attention_dim'] = 1024
         unet = UNet2DConditionModel(**config[0])
 
-    # Prevent fsdp from wrapping up_blocks and down_blocks because the forward pass calls length on these
-    unet.up_blocks._fsdp_wrap = False
-    unet.down_blocks._fsdp_wrap = False
-    for block in unet.up_blocks:
-        block._fsdp_wrap = True
-    for block in unet.down_blocks:
-        block._fsdp_wrap = True
-
-    if encode_latents_in_fp16:
-        vae = AutoencoderKL.from_pretrained(vae_model_name, torch_dtype=torch.float16)
-        text_encoder = CLIPTextModel.from_pretrained(model_name, subfolder='text_encoder', torch_dtype=torch.float16)
-    else:
-        vae = AutoencoderKL.from_pretrained(vae_model_name)
-        text_encoder = CLIPTextModel.from_pretrained(model_name, subfolder='text_encoder')
+        # Zero initialization trick
+        for name, layer in unet.named_modules():
+            # Final conv in ResNet blocks
+            if name.endswith('conv2'):
+                layer = zero_module(layer)
+            # proj_out in attention blocks
+            if name.endswith('to_out.0'):
+                layer = zero_module(layer)
+        # Last conv block out projection
+        unet.conv_out = zero_module(unet.conv_out)
+
+    torch_dtype = torch.float16 if encode_latents_in_fp16 else None
+    try:
+        vae = AutoencoderKL.from_pretrained(vae_model_name, subfolder='vae', torch_dtype=torch_dtype)
+    except:  # for handling SDXL vae fp16 fixed checkpoint
```

Comment:
The blanket except isn't great here. We should probably qualify the exception type here.